tests/Resources/ti.ui.window.test.js

```diff
@@ -96,6 +96,146 @@ describe('Titanium.UI.Window', function () {
 		});
 	});
 
+	it.ios('.leftNavButtons and .rightNavButtons', finish => {
+		const rightButton1 = Ti.UI.createButton({
+			title: 'Right1',
+			color: 'green',
+		});
+
+		const rightButton2 = Ti.UI.createButton({
+			title: 'Right2',
+			color: 'green',
+		});
+
+		const leftButton = Ti.UI.createButton({
+			title: 'Left',
+			color: 'blue'
+		});
+
+		const rootWindow = Ti.UI.createWindow({
+			backgroundColor: 'white',
+			leftNavButtons: [ leftButton ],
+			rightNavButtons: [ rightButton1, rightButton2 ],
+		});
+
+		win = Ti.UI.createNavigationWindow({
+			window: rootWindow
+		});
+
+		win.open();
+
+		rootWindow.addEventListener('focus', function focus() {
+			rootWindow.removeEventListener('focus', focus);
+			try {
+				should(rootWindow.rightNavButtons).be.an.Array();
+				should(rootWindow.rightNavButtons.length).be.eql(2);
+				rootWindow.rightNavButtons = [ rightButton1 ];
+				should(rootWindow.rightNavButtons.length).be.eql(1);
+
+				should(rootWindow.leftNavButtons).be.an.Array();
+				should(rootWindow.leftNavButtons.length).be.eql(1);
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+	});
+
+	it.ios('.leftNavButton and .rightNavButton  with color', finish => {
+		this.timeout(10000);
+
+		const nonColorButton = Ti.UI.createButton({
+			title: 'Right',
+		});
+
+		const greenColorButton = Ti.UI.createButton({
+			title: 'Right',
+			color: 'green',
+		});
+
+		const leftButton = Ti.UI.createButton({
+			title: 'Left',
+			color: 'blue'
+		});
+
+		const rootWindow = Ti.UI.createWindow({
+			backgroundColor: 'white',
+			leftNavButton: leftButton,
+			rightNavButton: nonColorButton,
+		});
+
+		win = Ti.UI.createNavigationWindow({
+			window: rootWindow
+		});
+
+		win.open();
+
+		rootWindow.addEventListener('postlayout', function postlayout() {
+			rootWindow.removeEventListener('postlayout', postlayout);
+			setTimeout(function () {
+				const nonColorButtonImage = win.toImage();
+				rootWindow.rightNavButton = greenColorButton;
+				setTimeout(function () {
+					try {
+						should(rootWindow.leftNavButton).be.an.Object();
+						should(rootWindow.rightNavButton).be.an.Object();
+						should(win).not.matchImage(nonColorButtonImage, 0); //  Navbutton without color should be different with a colored one
+					} catch (e) {
+						return finish(e);
+					}
+					finish();
+				}, 1000);
+			}, 1000);
```

Comment:
why the 1 second delay? Why not use a much shorter delay? (same in the line above)

Reply:
Let me try with shorter delay.